tests/integration_tests/utils.py

```diff
@@ -683,3 +686,106 @@ def train_with_backend(
     finally:
         # Remove results/intermediate data saved to disk
         shutil.rmtree(output_dir, ignore_errors=True)
+
```

Comment:
just a general question so I understand a bit more context, are we saying these setup functions should be generic enough to be reused across the different types of features , do we need to refactor the other tests to use this?

Reply:
I'm hoping the two setup functions can be used by any input or output feature test.   Some refactoring may be needed for the output setup when we complete the output sequence feature migration.

To use these functions in other feature unit tests, some refactoring will need to be done.  I'm hoping it will be a net decrease on LOC for the particular unit test.  

I think you may have worked on the `test_binary.py` test.  So you can see the difference.  Though in that particular one, there was only one test case, basically for the binary input feature.  I had to tweak the code in that.  I added the second function to test the binary output feature.